File: business-easy-core/src/main/java/com/businesseasy/core/entities/AccountEntity.java
package com.businesseasy.core.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "account", uniqueConstraints = {@UniqueConstraint(columnNames = {"bank", "account_no"})})
public class AccountEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", updatable = false, nullable = false)
    Long id;

    @Column(name = "account_name")
    String accountName;

    @Column(name = "holder_name")
    String holderName;

    @Column(name = "bank")
    String bank;

    @Column(name = "branch")
    String branch;

    @Column(name = "account_no")
    String accountNo;

    @Column(name = "balance")
    BigDecimal balance;
}

File: easiness-ui/app/server/entity/account.entity.ts
import * as Big from "big.js"
import { Entity, PrimaryGeneratedColumn, Column, Unique } from "typeorm"
import { AccountType } from "../model/account.model"
import { Base } from "./base.entity"

export const uniqueAccountCols = ['accountNo', 'bank']

@Entity({ name: 'account' })
@Unique(uniqueAccountCols)
export class Account extends Base {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    type: AccountType

    @Column({ unique: true })
    accountName: string

    @Column()
    holderName: string

    @Column({ nullable: true })
    bank!: string

    @Column({ nullable: true })
    branch!: string

    @Column()
    accountNo: string

    @Column('numeric', {
        precision: 20,
        scale: 6,
        transformer: {
            from: (value: string) => new Big(value),
            to: (value: Big) => value?.toString()
        }
    })
    balance: Big
}